### PR TITLE
Custom providers

### PR DESCRIPTION
**Pull request by markcowl**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/CustomProviders/tools/Resources/.gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+* text=auto
```
Comment (dolauli) [inline src/CustomProviders/tools/Resources/.gitattributes:1]: The **tools** folder is not required to be checked in.

File: src/CustomProviders/examples/New-AzCustomProvider.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Create a custom resource provider
+
+### Example 2: Create a custom provider with associations
+```powershell
+PS C:\> New-AzCustomProvider -ResourceGroupName myRG -Name Namespace.Type -Location "West US 2" -ResourceType @{Name="CustomRoute1"; Endpoint="https://www.contoso.com/"}, @{Name="Associations"; Endpoint="https://contoso.com/myService", RoutingType="Proxy,Cache,Extension"}
+
+Location  Name             Type
+--------  ----             ----
+West US 2 Namespace2.Type   Microsoft.CustomProviders/resourceproviders
```
Comment (dcaro) [inline src/CustomProviders/examples/New-AzCustomProvider.md:19]: It looks like the name returned is non consistent with the the command

File: src/CustomProviders/examples/Get-AzCustomProvider.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+```powershell
+PS C:\> Get-AzCustomProvider -ResourceGroupName myRg -Name Namespace.Type | Format-List
+
+Action            :
+Id                : /subscriptions/xxxxx-yyyyy-xxxx-yyyy/resourceGroups/mc-cp01/providers/Microsoft.CustomProviders/resourceproviders/Namespace.Type
+Location          : West US 2
+Name              : Namespace.Type
+ProvisioningState : Succeeded
+ResourceType      : {CustomRoute1, associations}
+Tag               : Microsoft.Azure.PowerShell.Cmdlets.CustomProviders.Models.Api20180901Preview.ResourceTags
```
Comment (dcaro) [inline src/CustomProviders/examples/Get-AzCustomProvider.md:23]: Should we have the tag list here?

File: src/CustomProviders/examples/Update-AzCustomProvider.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+```powershell
+PS C:\> Update-AzCustomProvider -ResourceGroupName myRg -Name Namespace.Type -Tag @{MyTag="MyValue"} | Format-List
+
+Action            :
+Id                : /subscriptions/xxxxx-yyyyy-xxxx-yyyy/resourceGroups/mc-cp01/providers/Microsoft.CustomProviders/resourceproviders/Namespace.Type
+Location          : West US 2
+Name              : Namespace.Type
+ProvisioningState : Succeeded
+ResourceType      : {CustomRoute1, associations}
+Tag               : Microsoft.Azure.PowerShell.Cmdlets.CustomProviders.Models.Api20180901Preview.ResourceTags
```
Comment (dcaro) [inline src/CustomProviders/examples/Update-AzCustomProvider.md:11]: Isn't his supposed to be the tag values ?

File: src/CustomProviders/examples/Get-AzCustomProvider.md (hunk 3 above)
Comment (isra-fel) [reply src/CustomProviders/examples/Get-AzCustomProvider.md:23]: Agreed. It's a design flaw and I believe there's currently no workaround. 
I've logged an issue here: https://github.com/Azure/autorest.powershell/issues/618

File: src/CustomProviders/examples/New-AzCustomProvider.md (hunk 2 above)
Comment (isra-fel) [reply src/CustomProviders/examples/New-AzCustomProvider.md:19]: Eagle eye. Fixed

**Pull request merged**